src/water_cmd.cpp

```diff
@@ -1361,11 +1361,42 @@ static VehicleEnterTileStatus VehicleEnter_Water(Vehicle *v, TileIndex tile, int
 	return VETSB_CONTINUE;
 }
 
+/**
+ * Check whether a tile with or without a flooding behaviour can be terraformed.
+ *
+ * @param tile Tile to check.
+ * @param z_new The new height the tile would get after being terraformed.
+ * @param tileh_new The new slope type the tile would get after being terraformed.
+ * @return true iff a tile with or without a flooding behaviour can be terraformed.
+ * @note This function is intended to be used for the town growth algorithm using the terraform command.
+ */
+bool CanTerraformFloodingTile(TileIndex tile, int z_new, Slope tileh_new)
```

Comment:
It might make more sense for this to live in `town_cmd.cpp`.

```suggestion
bool CanTownTerraformTileWithoutFlooding(TileIndex tile, int z_new, Slope tileh_new)
```

Reply:
That would add extra includes.

Moving it to town_cmd.cpp would require me to add an include "water.h" to reach GetFloodingBehaviour. Then I'm not really sure where to declare CanTownTerraformTileWithoutFlooding. clear_cmd.cpp, water_cmd.cpp and tree_cmd.cpp could need more includes too.

The way it is right now, everything is reached, without needing inclusions.

Reply:
I decided to move `CanTownTerraformTileWithoutFlooding` to `town_cmd.cpp` after all, because a rebase was now requiring me to add an include in `clear_cmd.cpp` when it wasn't needed previously. I took this opportunity to actually do the right thing, and ended up doing as @2TallTyler suggested, move code around.